### ISSUE TITLE
Shouldn't allow overriding a class method with an instance method

### ISSUE THREAD

**Issue by gvanrossum** [python/mypy#1867]
See #1866. We had a bug where a subclass had omitted the `@classmethod` decorator on a method it was overriding. Yet mypy did not flag this as an error. Smaller repro:

``` python
class C:
    @classmethod
    def foo(cls) -> None: pass
class D(C):
    def foo(self) -> None: pass
```

This code is IMO unsound in the presence of Type[C] -- D is a subclass of C but we cannot call .foo() on the subclass without an instance. (The actual example didn't use Type[C], it used dynamic dispatch.)

**Comment by gnprice** [comment]
Guido suspects there's a reason this isn't already an error -- so there may be discussion needed here once someone digs in to understand it better.

**Comment by gvanrossum** [comment]
It's not a very strong suspicion though. It could just be an oversight;
maybe class methods were added later and this was missed. Let's fix this
and see what it catches.

--Guido (mobile)

**Comment by afrieder** [comment]
I have a fix for this, but the stdlib actually uses this. importlib has numerous classmethods that override non-classmethods (for example, in [importlib/machinery.pyi](https://github.com/python/typeshed/blob/master/stdlib/3/importlib/machinery.pyi), all of those subclass [MetaPathFinder](https://github.com/python/typeshed/blob/master/stdlib/3/importlib/abc.pyi) which doesn't have classmethods.) Though looking at the actual [CPython implementation](https://github.com/python/cpython/blob/master/Lib/importlib/_bootstrap.py), I don't see them _actually_ subclassing the ABCs, so it might be a typeshed mistake? I'd appreciate some guidance here.

**Comment by ilevkivskyi** [comment]
@afrieder I think it is safe to override instance methods with class methods (and this is what you see in stubs) because one can call class methods on instances. Only the opposite is unsafe (as originally posted): overriding class methods with instance methods.
